core/os/os2/file_linux.odin

```diff
@@ -6,41 +6,32 @@ import "core:time"
 import "base:runtime"
 import "core:sys/linux"
 
+File_Impl_Kind :: enum u8 {
```

Comment:
I think the general idea for that `enum` was more for future use.  I will just take it out.  Why do we care if `stdin` is a pipe?

Reply:
It might be a file, in which case a program could attempt to read it's size. The `enum` was made is already available in the `master` branch and `os2.stat` should output the type of the handle. I'm not sure if your PR has my file_type changes, and if not i'd suggest rebasing off of master and see how I'm doing it right now (in short that information is not stored), however the change where you return a `.No_Size` error is still good.

Reply:
Oh, now I remember, why I added that.  I forgot to actually implement the other side of it.  It was to prevent non-blocking reads from returning EOF all the time.

Reply:
I'll have to do some research to know for sure whether there are any better solutions to this problem, but can you provide the setup where returning EOF is a problem?

Something like setting up the pipe and calling `read_entire_file`? You say non-blocking and I'm a little bit confused by that, clearly you mean `O_NONBLOCK` ops, because for now we don't have a way to queue up asynchronous operations, but why are pipes non-blocking? My perception is that this should apply to all kinds of files, potentially, be it a regular file on a network drive, or a named pipe in the filesystem.

Reply:
Indeed, I was thinking about `read_entire_file`.  I have personally used it after modifying termios, so that is sort of outside the scope of core anyway.  Is it really a problem?  No, in practice, it can always be mitigated.